crates/core_arch/src/simd.rs

```diff
@@ -3,25 +3,22 @@
 #![allow(non_camel_case_types)]
 
 macro_rules! simd_ty {
-    ($id:ident [$ety:ident]: $($elem_ty:ident),* | $($elem_name:ident),*) => {
+    ($id:ident: [$ety:ident; $ecount:literal]) => {
         #[repr(simd)]
         #[derive(Copy, Clone, Debug, PartialEq)]
-        pub(crate) struct $id($(pub $elem_ty),*);
+        pub(crate) struct $id(pub [$ety; $ecount]);
 
         #[allow(clippy::use_self)]
         impl $id {
             #[inline(always)]
-            pub(crate) const fn new($($elem_name: $elem_ty),*) -> Self {
-                $id($($elem_name),*)
+            pub(crate) const fn new(val: [$ety; $ecount]) -> Self {
+                $id(val)
```

Comment:
From what I've heard, the plan is to make any direct field accesses (including calling the constructor like this) illegal on `#[repr(simd)]` types and only allow creation of the type and access to the inner array through `transmute` or pointer operations, so I think it might make sense to make the struct field private and to do something similar to rust-lang/portable-simd#339 or rust-lang/portable-simd#342 in `new` and `splat`.

Reply:
I think it'd also be fine for that to be a separate PR, if you prefer.

Reply:
> direct field accesses (including calling the constructor like this) illegal

Emphasis on *plan*. This isn't happening with the current MCP, we are just removing the field-style `#[repr(simd)]`. In the longer term, I think scottmcm and other people have plans for that but not immediately.